File: python/armstrong-numbers/armstrong_numbers.py
def get_digits(number):
    return list(map(int, list(str(number))))


def is_armstrong(number):
    digits = get_digits(number)
    num_digits = len(digits)
    digits_raised_to_num_digits = [digit ** num_digits for digit in digits]

    return number == sum(digits_raised_to_num_digits)

File: go/armstrong-numbers/armstrong_numbers.go
package armstrong

import (
	"fmt"
	"math"
	"strconv"
)

func IsNumber(n int) bool {
	sum := 0
	digits := fmt.Sprint(n)
	for _, digit := range digits {
		d, err := strconv.Atoi(string(digit))
		if err != nil {
			panic(err)
		}
		sum += intPow(d, len(digits))
	}
	return sum == n
}

func intPow(n int, power int) int {
	return int(math.Pow(float64(n), float64(power)))
}
